test/e2e/live_iso_test.go

```diff
@@ -0,0 +1,95 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	infrav1 "github.com/metal3-io/cluster-api-provider-metal3/api/v1beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("When testing live iso [live-iso]", func() {
+	liveIsoTest()
+})
+
+func liveIsoTest() {
+	BeforeEach(func() {
+		validateGlobals(specName)
+		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
+		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+	})
+
+	It("Should create a target cluster node with live iso", func() {
+		controlPlaneMachineCount = 1
+		workerMachineCount = 1
+
+		workloadClusterTemplate := clusterctl.ConfigCluster(ctx, clusterctl.ConfigClusterInput{
+			KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+			ClusterctlConfigPath:     clusterctlConfigPath,
+			Flavor:                   "live-iso",
+			Namespace:                namespace,
+			ClusterName:              clusterName,
+			KubernetesVersion:        e2eConfig.GetVariable(KubernetesVersion),
+			ControlPlaneMachineCount: &controlPlaneMachineCount,
+			WorkerMachineCount:       &workerMachineCount,
+			InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+			LogFolder:                clusterctlLogFolder,
+		})
+
+		Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
+		Logf("Applying the cluster template yaml to the cluster")
+		Expect(bootstrapClusterProxy.Apply(ctx, workloadClusterTemplate)).To(Succeed())
+
+		var controlplaneM3Machines, workerM3Machines []infrav1.Metal3Machine
+
+		By("Waiting for the control plane and worker node to exist")
```

Comment:
```suggestion
		By("Waiting for the control plane and worker metal3machines to exist")
```

Reply:
I would like to have @Xenwar input on this. To my knowledge for this test in cluster we need at least one node to be using live iso.

Reply:
I agree with @lentzi90 on
`Alternatively, we could just apply/patch a BMH to use live-iso to boot and be done with it.`
But, does the e2e code allows us to just provisioning a node, without setting up a cluster ?

Reply:
I think it should be possible. What I can see now in the live-iso test we start by creating a target cluster. So this means the bootstrap cluster is already there and it should also have the BMHs in available state.
Now we should be able to manipulate the BMH in a similar way like it is done in the remediation test where they [add annotations](https://github.com/metal3-io/cluster-api-provider-metal3/blob/548790c0acba5c10cdaf537e1e7fdbd9e80501ca/test/e2e/common.go#L109).